firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/firestore.kt

```diff
@@ -392,19 +393,83 @@ class FirebaseFirestoreTest {
         }
 
         val data = DataWithDocumentReference(documentRef1)
-        // store geo point
-        getDocument().set(data)
+        // store reference
+        getDocument().set(DataWithDocumentReference.serializer(), data)
         // restore data
-        val savedData = getDocument().get().data<DataWithDocumentReference>()
+        val savedData = getDocument().get().data(DataWithDocumentReference.serializer())
         assertEquals(data.documentReference.path, savedData.documentReference.path)
 
         // update data
         val updatedData = DataWithDocumentReference(documentRef2)
         getDocument().update(
-            FieldPath(DataWithDocumentReference::documentReference.name) to updatedData.documentReference.withSerializer(FirebaseDocumentReferenceSerializer)
+            FieldPath(DataWithDocumentReference::documentReference.name) to updatedData.documentReference.withSerializer(DocumentReferenceSerializer)
         )
         // verify update
-        val updatedSavedData = getDocument().get().data<DataWithDocumentReference>()
+        val updatedSavedData = getDocument().get().data(DataWithDocumentReference.serializer())
         assertEquals(updatedData.documentReference.path, updatedSavedData.documentReference.path)
     }
+
+    @Serializable
+    data class TestDataWithDocumentReference(
+        val uid: String,
+        val reference: DocumentReference
```

Comment:
Can you also assert/test if optional reference encoded/decoded successfully?

Reply:
done